lib/block_view.go

```diff
@@ -3515,40 +3576,121 @@ func (bav *UtxoView) Preload(desoBlock *MsgDeSoBlock, blockHeight uint64) error
 	return nil
 }
 
-func (bav *UtxoView) GetNextNonceForPublicKey(pkBytes []byte) (uint64, error) {
-	pkidEntry := bav.GetPKIDForPublicKey(pkBytes)
-	if pkidEntry == nil || pkidEntry.isDeleted {
-		return 0, fmt.Errorf("GetNextNonceForPublicKey: PKID entry is deleted for public key %s", PkToStringBoth(pkBytes))
+func (bav *UtxoView) GetTransactorNonceEntry(nonce *DeSoNonce, pkid *PKID) (*TransactorNonceEntry, error) {
+	if nonce == nil {
+		return nil, fmt.Errorf("GetTransactorNonceEntry: nil nonce")
+	}
+	if pkid == nil {
+		return nil, fmt.Errorf("GetTransactorNonceEntry: nil pkid")
+	}
+	mapKey := TransactorNonceMapKey{
+		Nonce:          *nonce,
+		TransactorPKID: *pkid,
 	}
-	return bav.GetNextNonceForPKID(*pkidEntry.PKID)
-}
 
-func (bav *UtxoView) GetNextNonceForPKID(pkid PKID) (uint64, error) {
-	nextNonce, nonceFound := bav.PKIDToNextNonce[pkid]
-	if nonceFound {
-		return nextNonce, nil
+	nonceEntry, exists := bav.TransactorNonceMapKeyToTransactorNonceEntry[mapKey]
+	if exists {
```

Comment:
nit: I usually do a `if exists && nonceEntry != nil` check here. I dunno how nonceEntry would ever be nil, but double-checking just in case fwiw.